api/v1alpha4/conversion.go

```diff
@@ -107,3 +109,123 @@ func (r *OpenStackMachineTemplateList) ConvertFrom(srcRaw ctrlconversion.Hub) er
 
 	return Convert_v1beta1_OpenStackMachineTemplateList_To_v1alpha4_OpenStackMachineTemplateList(src, r, nil)
 }
+
+func Convert_v1alpha4_SubnetFilter_To_v1beta1_SubnetFilter(in *SubnetFilter, out *v1beta1.SubnetFilter, s conversion.Scope) error {
+	out.Name = in.Name
+	out.Description = in.Description
+	if in.ProjectID != "" {
+		out.ProjectID = in.ProjectID
+	} else {
+		out.ProjectID = in.TenantID
+	}
+	out.IPVersion = in.IPVersion
+	out.GatewayIP = in.GatewayIP
+	out.CIDR = in.CIDR
+	out.IPv6AddressMode = in.IPv6AddressMode
+	out.IPv6RAMode = in.IPv6RAMode
+	out.ID = in.ID
+	out.Tags = in.Tags
+	out.TagsAny = in.TagsAny
+	out.NotTags = in.NotTags
+	out.NotTagsAny = in.NotTagsAny
+	return nil
+}
+
+func Convert_v1beta1_SubnetFilter_To_v1alpha4_SubnetFilter(in *v1beta1.SubnetFilter, out *SubnetFilter, s conversion.Scope) error {
+	out.TenantID = in.ProjectID
+	return autoConvert_v1beta1_SubnetFilter_To_v1alpha4_SubnetFilter(in, out, s)
+}
+
+func Convert_v1alpha4_Filter_To_v1beta1_NetworkFilter(in *Filter, out *v1beta1.NetworkFilter, s conversion.Scope) error {
+	out.Name = in.Name
+	out.Description = in.Description
+	if in.ProjectID != "" {
+		out.ProjectID = in.ProjectID
+	} else {
+		out.ProjectID = in.TenantID
+	}
+	out.ID = in.ID
+	out.Tags = in.Tags
+	out.TagsAny = in.TagsAny
+	out.NotTags = in.NotTags
+	out.NotTagsAny = in.NotTagsAny
+	return nil
+}
+
+func Convert_v1beta1_NetworkFilter_To_v1alpha4_Filter(in *v1beta1.NetworkFilter, out *Filter, s conversion.Scope) error {
+	out.Name = in.Name
+	out.Description = in.Description
+	out.ProjectID = in.ProjectID
+	out.TenantID = in.ProjectID
+	out.ID = in.ID
+	out.Tags = in.Tags
+	out.TagsAny = in.TagsAny
+	out.NotTags = in.NotTags
+	out.NotTagsAny = in.NotTagsAny
+	return nil
+}
+
+func Convert_v1alpha4_PortOpts_To_v1beta1_PortOpts(in *PortOpts, out *v1beta1.PortOpts, s conversion.Scope) error {
+	err := autoConvert_v1alpha4_PortOpts_To_v1beta1_PortOpts(in, out, s)
+	if err != nil {
+		return err
+	}
```

Comment:
This could be written in short syntax.
We should probably think about adding the [ifshort golangci linter](https://github.com/esimonov/ifshort)

Reply:
Do you know how to do that?

Reply:
fun fact: [it's already enabled](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/.golangci.yml#L28). Are we not calling the golangci-linter with our presubmit prow jobs?